bigchaindb/web/websocket_server.py

```diff
@@ -0,0 +1,174 @@
+"""WebSocket server for the BigchainDB Event Stream API."""
+
+# NOTE
+#
+# This module contains some functions and utilities that might belong to other
+# modules. For now, I prefer to keep everything in this module. Why? Because
+# those functions are needed only here.
+#
+# When we will extend this part of the project and we find that we need those
+# functionalities elsewhere, we can start creating new modules and organizing
+# things in a better way.
+
+
+import json
+import asyncio
+import logging
+import threading
+from uuid import uuid4
+
+import aiohttp
+from aiohttp import web
+
+from bigchaindb import config
+from bigchaindb.events import EventTypes
+
+
+logger = logging.getLogger(__name__)
+POISON_PILL = 'POISON_PILL'
+EVENTS_ENDPOINT = '/api/v1/streams/valid_tx'
+
+
+def _put_into_capped_queue(queue, value):
+    """Put a new item in a capped queue.
+
+    If the queue reached its limit, get the first element
```

Comment:
Uhm, I see what you mean. I don't have a good answer for that. My assumption is that creating blocks is computationally more intense than dispatching messages to the connected clients, but I've never tested it. Let's address this issue when we have more tools to do stress testing.